lib/Sema/CSSimplify.cpp

```diff
@@ -11518,6 +11663,36 @@ ConstraintSystem::SolutionKind ConstraintSystem::simplifyFixConstraint(
   }
 
   case FixKind::AllowArgumentTypeMismatch: {
+    // If this is after the code completion location, give a fixed penalty
+    // with the same impact as that for the fix to ignore arguments after the
+    // completion position when matching arguments.
+    if (isForCodeCompletion()) {
+      auto getArgIndex = [&](ConstraintLocator *loc) -> Optional<unsigned> {
+        if (auto component = loc->findLast<LocatorPathElt::ApplyArgToParam>())
+          return component->getArgIdx();
+        return None;
+      };
+      auto completionInfo = getCompletionArgInfo(locator.getAnchor(), *this);
+      auto argIdx = getArgIndex(getConstraintLocator(locator));
+      if (argIdx && isAfterCompletionToken(completionInfo, *argIdx)) {
+        // Make sure we only penalize argument mismatches after the completion
+        // position once.
+        bool isFirst = llvm::all_of(getFixes(), [&](const ConstraintFix *fix) {
+          auto *fixLocator = fix->getLocator();
+          if (fixLocator->getAnchor() != locator.getAnchor())
```

Comment:
I think you'd want to check the kind of fix here as well, otherwise it's possible to find some unrelated fix that has `ApplyArgToParam` in the path and use it.